### PR TITLE
Fix RowRecord to allow read partition values

### PR DESCRIPTION
**Pull request by scottsand-db**
This PR fixes a lacking feature in the RowRecord implementation where partition (metadata) values were not included and able to be read, only data values were.

### CODE REVIEW

File: standalone/src/main/scala/io/delta/standalone/internal/data/CloseableParquetDataIterator.scala [hunk 1]
```diff
@@ -116,12 +126,53 @@ private[internal] case class CloseableParquetDataIterator(
...
+    if (null != nextPartitionVals) {
+      nextPartitionVals.foreach { case (fieldName, value) =>
+        if (value == null || "null" == value) {
+          partitionValues += (fieldName -> null)
+        } else {
+          val schemaField = schema.get(fieldName)
+          if (schemaField != null) {
+            val decodedFieldValue = decodePartition(schemaField.getDataType, value)
+            partitionValues += (fieldName -> decodedFieldValue)
+          }
```
Comment (zsxwing) [inline standalone/src/main/scala/io/delta/standalone/internal/data/CloseableParquetDataIterator.scala:147]: it's better to throw an exception for the invalid state (`schemaField==null`).

File: standalone/src/main/scala/io/delta/standalone/internal/data/CloseableParquetDataIterator.scala [hunk 2]
```diff
@@ -116,12 +126,53 @@ private[internal] case class CloseableParquetDataIterator(
...
+   * @return the iterable for the next data file in `dataFilePathsAndPartitionsIter`, not null
    */
   private def readNextFile: ParquetIterable[RowParquetRecord] = {
+    val (nextDataFilePath, nextPartitionVals) = dataFilePathsAndPartitionsIter.next()
+
+    partitionValues = Map()
+
+    if (null != nextPartitionVals) {
+      nextPartitionVals.foreach { case (fieldName, value) =>
+        if (value == null || "null" == value) {
```
Comment (zsxwing) [inline standalone/src/main/scala/io/delta/standalone/internal/data/CloseableParquetDataIterator.scala:140]: I missed this in my previous review. `"null"` should be a string `"null"`.
```
if (value == null) {
  partitionValues += (fieldName -> null)
}
```
Could you update the test to test the string `"null"` case?

File: standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala [hunk 3]
```diff
@@ -82,9 +84,18 @@ private[internal] case class RowParquetRecordImpl(
 
   override def getSchema: StructType = schema
 
-  override def getLength: Int = record.length
+  override def getLength: Int = record.length + partitionValues.size
 
-  override def isNullAt(fieldName: String): Boolean = record.get(fieldName) == NullValue
+  override def isNullAt(fieldName: String): Boolean = {
+    if (partitionValues.contains(fieldName)) { // is partition field
+        if (partitionValues(fieldName) == null) {
```
Comment (zsxwing) [inline standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala:91]: nit: `return partitionValues(fieldName) == null`

File: standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala [hunk 4]
```diff
@@ -133,6 +144,11 @@ private[internal] case class RowParquetRecordImpl(
    */
   private def getAs[T](fieldName: String): T = {
     val schemaField = schema.get(fieldName)
+
+    if (partitionValues.contains(fieldName)) { // partition field
+      if (partitionValues(fieldName) == null) return null.asInstanceOf[T]
```
Comment (zsxwing) [inline standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala:149]: do we really need this line?

**Review by zsxwing — COMMENTED** [review]
Left some minor comments

File: standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala (hunk 4 above)
Comment (scottsand-db) [reply standalone/src/main/scala/io/delta/standalone/internal/data/RowParquetRecordImpl.scala:149]: good catch sorry

**Review by zsxwing — COMMENTED** [review]
LGTM. Manually merged: https://github.com/delta-io/connectors/commit/117e7820467f2c5b801373974e10e3e6fb77eb5c

**Pull request closed without merging**